src/pythie_serving/table_wrapper.py

```diff
@@ -0,0 +1,82 @@
+import os
+import logging
+
+import grpc
+import numpy as np
+import json
+
+from .tensorflow_proto.tensorflow_serving.config import model_server_config_pb2
+from .tensorflow_proto.tensorflow_serving.apis import predict_pb2, prediction_service_pb2_grpc
+from .utils import make_ndarray_from_tensor
+from .exceptions import PythieServingException
+import csv
+
+
+class TablePredictionServiceServicer(prediction_service_pb2_grpc.PredictionServiceServicer):
+
+    def __init__(self, *, logger: logging.Logger, model_server_config: model_server_config_pb2.ModelServerConfig):
+        self.logger = logger
+        self.table_map = {}
+        for table_config in model_server_config.model_config_list.config:
+            with open(os.path.join(table_config.base_path, "metadata.json"), "r") as f:
+                metadata = json.load(f)
+
+            with open(os.path.join(table_config.base_path, table_config.name) + ".csv", mode='r') as csvfile:
+                reader = csv.DictReader(csvfile)
+                table_as_dict = [row for row in reader]
+                self.table_map[table_config.name] = {'table': table_as_dict}
+
+            self.table_map[table_config.name] = {
+                "table": table_as_dict,
+                "feature_names": metadata["feature_names"],
+                "nb_features": len(metadata["feature_names"]),
+                "target_name": metadata["target_name"],
+            }
+
+    def Predict(self, request: predict_pb2.PredictRequest, context: grpc.RpcContext):
+        table_name = request.model_spec.name
+        if table_name not in self.table_map:
+            raise PythieServingException(f'Unknown table: {table_name}. This pythie-serving instance can only '
+                                         f'serve one of the following: {",".join(self.table_map.keys())}')
+
+        table_dict = self.table_map[table_name]
+        table = table_dict["table"]
+        features_names = table_dict["feature_names"]
+        nb_features = table_dict["nb_features"]
+        target_name = table_dict["target_name"]
+
+        for feature_name in features_names:
+            if feature_name not in request.inputs:
+                raise PythieServingException(f"{feature_name} not set in the predict request.")
+
+        nb_samples = request.inputs[features_names[0]].tensor_shape.dim[0].size
+        samples = np.empty((nb_samples, nb_features), object)
+
+        for feature_index, feature_name in enumerate(features_names):
+
+            if request.inputs[features_names[0]].tensor_shape.dim[0].size != nb_samples:
+                raise PythieServingException(f"{feature_name} has invalid length.")
+
+            nd_array = make_ndarray_from_tensor(request.inputs[feature_name])
```

Comment:
this 3 lines should be factorized between all PredictionServiceServicer i think

Reply:
Which 3 lines ? Do you mean has as function in utils.py ?

Reply:
yes 
```
            nd_array = make_ndarray_from_tensor(request.inputs[feature_name])
            if len(nd_array.shape) != 2 or nd_array.shape[1] != 1:
                raise PythieServingException('All input vectors should be 1D tensor')
```  are basically called in every wrapper

Reply:
Ok, in that case it would make more sense to factorize all 3 checks (feature_name not set in the predict request, feature_name has invalid length, All input vectors should be 1D tensor) in utils.py, no? I can do that